frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/StaticEmbedSetupPane.tsx

```diff
@@ -174,48 +183,90 @@ export const StaticEmbedSetupPane = ({
 
       <Tabs defaultValue={TABS.Overview} data-testid="embedding-preview">
         <Tabs.List p="0 1.5rem">
-          <Tabs.Tab value={TABS.Overview}>{t`Overview`}</Tabs.Tab>
-          <Tabs.Tab value={TABS.Parameters}>{t`Parameters`}</Tabs.Tab>
-          <Tabs.Tab value={TABS.Appearance}>{t`Appearance`}</Tabs.Tab>
+          <Tabs.Tab
+            value={TABS.Overview}
+            onClick={() => setActiveTab(TABS.Overview)}
+          >{t`Overview`}</Tabs.Tab>
+          <Tabs.Tab
+            value={TABS.Parameters}
+            onClick={() => setActiveTab(TABS.Parameters)}
+          >{t`Parameters`}</Tabs.Tab>
+          <Tabs.Tab
+            value={TABS.Appearance}
+            onClick={() => setActiveTab(TABS.Appearance)}
+          >{t`Appearance`}</Tabs.Tab>
         </Tabs.List>
-        <Tabs.Panel value={TABS.Overview}>
-          <OverviewSettings
-            resourceType={resourceType}
-            selectedServerCodeOption={selectedServerCodeOption}
-            serverEmbedCodeSlot={getServerEmbedCodePane(TABS.Overview)}
-          />
-        </Tabs.Panel>
-        <Tabs.Panel value={TABS.Parameters}>
-          <ParametersSettings
-            activePane={activePane}
-            resourceType={resourceType}
-            resourceParameters={resourceParameters}
-            embeddingParams={embeddingParams}
-            lockedParameters={lockedParameters}
-            parameterValues={parameterValues}
-            iframeUrl={iframeUrl}
-            displayOptions={displayOptions}
-            serverEmbedCodeSlot={getServerEmbedCodePane(TABS.Parameters)}
-            onChangeEmbeddingParameters={setEmbeddingParams}
-            onChangeParameterValue={(id: string, value: string) =>
-              setParameterValues(state => ({
-                ...state,
-                [id]: value,
-              }))
-            }
-            onChangePane={setActivePane}
-          />
-        </Tabs.Panel>
-        <Tabs.Panel value={TABS.Appearance}>
-          <AppearanceSettings
-            activePane={activePane}
-            resourceType={resourceType}
-            iframeUrl={iframeUrl}
-            displayOptions={displayOptions}
-            serverEmbedCodeSlot={getServerEmbedCodePane(TABS.Appearance)}
-            onChangePane={setActivePane}
-            onChangeDisplayOptions={setDisplayOptions}
-          />
+        <Tabs.Panel value={activeTab}>
```

Comment:
This is a bit hacky, but essentially, I need the preview to stay in the same component, otherwise, it will unmount and remount when changing between tabs because we have 3 parents (`Tabs.Panel`) previously. Now it's reduced to just 1.

Reply:
It took a bit to understand why this is working lol, even with the context of the pr and your comments, should we add a comment? Otherwise it could be very easy to think "why are they doing this??" when reading the code :D

Reply:
Yeah, good point, the comment should go into the code.

Reply:
Added a comment in https://github.com/metabase/metabase/pull/38271/commits/2b8172c10d0c10cf15f6aa25665b869f508ac84d